### ISSUE TITLE
Mo' Creatures Extended release Version 12.4.1 Interfering with Duplicate Vanilla Mob spawning

### ISSUE THREAD

**Issue by Catullus16** [Elite-Modding-Team/MoCreaturesExtended#133]
I've noticed an issue where certain mobs in the Mo' Creatures extended mod prevent their vanilla duplicates from spawning as well as duplicates from other mods that add mobs to the game with the same name. I am currently running Minecraft version 1.12.2 on Multi MC launcher with the following other mods that add extra mobs to the game:
Oceanic expanse version 1.0.7
Quark Version 1.6-179
Future MC Version 0.2.15

I can confirm spawning interference for at least the following duplicate mobs so far:

Wild Polar Bear: Duplicate with vanilla polar bears: Setting B:canSpawn=false also seems to disable the spawning of vanilla polar bears as well. Vanilla polar bears also fail to generate when B:canSpawn=true.

Bunny: Duplicate with vanilla rabbits: Both bunnies from Mo' Creatures Extended and vanilla rabbits generate at the same time when B:canSpawn=true. Setting to B:canSpawn=false also seems to disable the spawning of vanilla rabbits.

Kitty: Duplicate of vanilla cats: Normally vanilla cats spawn around villages when no interfering mods are installed. Setting B:canSpawn=false does not appear to allow vanilla cats to spawn normally around villages again.

Wild Horse: Duplicate of vanilla horse: Setting B:canSpawn=false also appears to disable the spawning of vanilla horses and vanilla horses fail to spawn when Mo' Creatures Extended is installed.

Crab: Duplicate with two other crab mobs installed: One from Quark and another from Oceanic Expanse. Both of these duplicate crab mobs fail to generate whether or not B:canSpawn in the cfg file is set to true of false.

The following mobs from Oceanic Expance and Future MC might also be suppressed when Mo' Creatures Extended is installed, but I have yet to encounter them frequently enough to confirm this:

Cod: duplicate with two other cod fish mobs: one from Future MC and another from Oceanic Expanse.

Salmon: duplicate with two other salmon mobs: one from Future MC and another from Oceanic Expanse.

Panda Bear: Duplicate with the panda mob from Future MC.

Bee: Duplicate with the Bee mob from Future MC. This may potentially interfere with the collection of honey in the future MC mod.

In the meantime I plan on creating separate world files to tests which of the other two mods may also be interfering with mob spawning along with Mo' Creatures extended. Let me know if you need a game log documenting the world generation process.

**Comment by ACGaming** [comment]
This doesn't make too much sense, I'm afraid. Even if spawn list entries would go by name, they are different for many listed entities:

`polar_bear` <-> `wildpolarbear`
`rabbit` <-> `bunny`
`ocelot` <-> `kitty`
`horse`<-> `wildhorse`
...

I also checked the spawnable lists of all biomes, only Mo' Creatures entities get disabled when modifying the `canSpawn` config values as intended. This isn't part of vanilla after all.

Curious, how did you come to this conclusion?

**Comment by Catullus16** [comment]
The other possibility is the vanilla variants have a lower mob spawn rate than the Mo' Creatures version. I need to do further testing to confirm this.

**Comment by Catullus16** [comment]
I will be generating some single-biome worlds this week to confirm if Mo' Creatures Extended is in fact interfering with vanilla mob spawn or if the issue is merely a lower mob spawn rate for vanilla mobs.

**Comment by Catullus16** [comment]
I did some world tests this evening using the Lonely Biomes mod to confirm whether or not Mo' Creatures was entirely suppressing the generation of vanilla mobs and is appears that the Mod is actually reducing the spawn rate of certain vanilla mobs rather than entirely suppressing their generation. I tested both by removing Mo' Creatures entirely and also just disabling the duplicate mods from Mo' Creatures.

 This is less of an issue for very common vanilla mobs, like rabbits, but it's more severe of an issue for vanilla mobs that already have a low spawn rate in the wild (ocelots, polar bears, donkeys). There currently does not appear to be an easy way to readjust the vanilla mod spawn rate to compensate without installing another external mod. I will attempt to find one and do further troubleshooting. Hopefully I don't have to learn a scripting language to operate the mod.
